utils/calculateBettingMoney.js

```diff
@@ -0,0 +1,201 @@
+const Game = require("../models/Game");
+const User = require("../models/User");
+const Statistic = require("../models/Statistic");
+const UserBettingData = require("../models/UserBettingData");
+
+const calculateLosingMoneyForWinner = async (winners, gameDate) => {
+  const gameData = await Game.findOne({ gameDate });
+  const totalBettingMoney = gameData.totalMoney / 10;
+  const winnerList = [];
+
+  winners.forEach((position) => {
+    const totalWinnerMoney = position.users
+      .flat()
+      .reduce((acc, winner) => acc + winner.bettingMoney, 0);
+    let losingMoney = totalBettingMoney - totalWinnerMoney;
+
+    position.users.forEach((group) => {
+      if (position.users.length > 1) {
+        losingMoney = Math.round(losingMoney / position.users.length);
+      }
+
+      group.forEach((user) => {
+        const totalWinnerMoneyPerGroup = group.reduce((acc, winner) => (
+          acc + winner.bettingMoney),
+        0);
+        const percentage = Number((user.bettingMoney / totalWinnerMoneyPerGroup).toFixed(2));
+        winnerList.push({
+          id: user.id,
+          earnedMoney: Math.round(totalWinnerMoney * percentage),
+        });
+      });
+    });
+  });
+
+  await Promise.all(
+    winnerList.map((user) => (
+      UserBettingData.findOneAndUpdate(
+        { gameDate, user: user.id },
+        {
+          $inc: {
+            earnedMoney: user.earnedMoney,
+          },
+        }
+      )
+    ))
+  );
+};
+
+const calculateBettingMoney = async (gameDate) => {
+  // 1. gameDate와 일치하는 Statistic을 찾음 : $match
+  // 2. score 점수로 오름차순 sorting : $sort
+  // 3. position 별로 그루핑 : $group
+  //    - "totalBettingMoney"의 값이 0 이상인 선수들만 $push
+  const statisticsPerPosition = await Statistic.aggregate([
+    {
+      $match: {
+        gameDate,
+      },
+    },
+    {
+      $sort: {
+        score: -1,
+      },
+    },
+    {
+      $group: {
+        _id: "$position",
+        players: {
+          $push: {
+            $cond: {
+              if: {
+                $gt: ["$totalBettingMoney", 0],
+              },
+              then: {
+                player: "$playerId",
+                score: "$score",
+                totalBettingMoney: "$totalBettingMoney",
+                users: "$users",
+              },
+              else: "$noval",
+            },
+          },
+        },
+      },
+    },
+  ]);
+
+  const selectedFirstPlayerUsers = [];
+  const selectedSecondPlayerUsers = [];
+  let awardedUsers = [];
+
+  statisticsPerPosition.forEach((position, posIndex) => {
```

Comment:
```suggestion
  statisticsPerPosition.forEach((position, posIndex) => {
```
posIndex 네이밍만 명확하게 수정 부탁드려용